docs/plans/2026-04-03-plugin-architecture-design.md

```
@@ -0,0 +1,192 @@
+New table:
+
+```sql
+CREATE TABLE IF NOT EXISTS schema_version (
+    version INTEGER NOT NULL,
+    applied_at TEXT NOT NULL DEFAULT (datetime('now'))
+);
+```
+
+Numbered migration files (embedded via `//go:embed`):
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)